src/main/resources/resources/cohortdefinition/sql/customEraStrategy.sql

```diff
@@ -3,9 +3,9 @@
 select de.PERSON_ID, DRUG_EXPOSURE_START_DATE,  COALESCE(DRUG_EXPOSURE_END_DATE, DATEADD(day,DAYS_SUPPLY,DRUG_EXPOSURE_START_DATE), DATEADD(day,1,DRUG_EXPOSURE_START_DATE)) as DRUG_EXPOSURE_END_DATE 
 INTO #drugTarget
 FROM @cdm_database_schema.DRUG_EXPOSURE de
-WHERE (de.drug_concept_id in (SELECT concept_id from  #Codesets where codeset_id = @drugCodesetId) 
```

Comment:
This change isn't yielding the same sub query results:  the logic is to only find drug exposures for people in the cohort (hence the ```where person_id in (select person id from @eventTable)```) .  The switch to using joins means that you get multiple rows per person in the event table in the result, and I'm not sure what impact that will have to the overall query...but it's not the same.

Reply:
It's possible (but I am still unhappy if we have to write query logic to cater to one specific cialect) that the above WHERE (A OR B) and C caould be rewritten as NOT (NOT A AND NOT B) AND C so the sql would be:

```
WHERE NOT (
  de.drug_concept_id NOT in (SELECT concept_id from  #Codesets where codeset_id = @drugCodesetId) 
  AND de.drug_source_concept_id NOT in (SELECT concept_id from  #Codesets where codeset_id = @drugCodesetId)
)  AND de.person_id in (select person_id from @eventTable)
```

Reply:
@chrisknoll maybe it's better to write something like this:
```FROM @cdm_database_schema.DRUG_EXPOSURE de
  WHERE de.drug_exposure_id IN (
    SELECT dei.drug_exposure_id FROM @cdm_database_schema.DRUG_EXPOSURE dei
      JOIN @eventTable events on events.person_id = de.person_id
      JOIN #Codesets cs on cs.codeset_id = @drugCodesetId AND
        (de.drug_concept_id = cs.concept_id OR de.drug_source_concept_id = cs.concept_id)
)
;```
i think it's more 'natural' and hope works same

Reply:
Does ```exists``` work in Impala
```
where exists (select 1 from FROM @cdm_database_schema.DRUG_EXPOSURE dei
      JOIN @eventTable events on events.person_id = de.person_id
      JOIN #Codesets cs on cs.codeset_id = @drugCodesetId AND
        (de.drug_concept_id = cs.concept_id OR de.drug_source_concept_id = cs.concept_id)
)
```